cgsmiles/drawing_utils.py

```diff
@@ -0,0 +1,193 @@
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap
+
+def angle_between(v1, v2):
+    cos_theta = np.dot(v1, v2) / (np.linalg.norm(v1) * np.linalg.norm(v2))
+    angle_rad = np.arccos(np.clip(cos_theta, -1.0, 1.0))
+    return np.degrees(angle_rad)
+
+def rotation_from_x_axis(v):
+    angle_rad = np.arctan2(v[1], v[0])
+    return np.degrees(angle_rad)
+
+def angle_of_interest(v1, v2):
+    vtot = v1 + v2
+    # Rotation angle with respect to the x-axis
+    angle_x_axis = rotation_from_x_axis(vtot)
+    return angle_x_axis
+
+def create_rwb_colormap(n):
+    cmap = plt.get_cmap('tab20', n)
+    colors = cmap(np.linspace(0, 1, n))
+    return ListedColormap(colors)
+
+default_colormap = create_rwb_colormap
+
+def rotate_2D(x, y, theta):
+    return x*math.cos(theta) - y*math.sin(theta), x*math.sin(theta) + y*math.cos(theta)
+
+def make_edge(p0, p1, bond_order, spacing=0.1, sep=0.0):
+    """
+    Given two positions `p0` and `p1` as well as a bond_order
+    generate a horizontal line or a double / triple line depending
+    on the bond order.
+
+    Parameters
+    ----------
+    p0: np.ndarray(2)
+    p1: np.ndarray(2)
+    bond_order: int
+    spacing: float
+    sep: float
+
+    Returns
+    -------
+    list(list([float, float]))
+        list of list of starting and end points for
+        the edge
+    """
+    # Step 1: Make a horizontal line from sep to bond_length - sep, and figure
+    #         out how far we'll need to rotate it. This line is (x0p, x1p)
+    #print(p0, p1)
+    x0, y0 = p0
+    x1, y1 = p1
+    bond_length = math.sqrt((x0 - x1)**2 + (y0-y1)**2)
+    sep = min(sep * bond_length, sep)
+    x0p = sep
+    x1p = bond_length - sep
+    theta = math.atan2((y1 - y0), (x1 - x0))
+
+    # Step 2: Split the line (if needed) in multiple copies (let's say 2). 1 up,
+    #         1 down. This results in horizontal line(s) ((x0p, y0p), (x1p, y1p)).
+    #         These we then rotate , resulting in ((x0pp, y0pp), (x1pp, y1pp)).
+    #         Finally, translate them from the origin to (x0, y0)
+    out = []
+    for f in range(bond_order):
+        y0p = y1p = spacing * ((1-bond_order)/2 + f)
+        x0pp, y0pp = rotate_2D(x0p, y0p, theta)
+        x1pp, y1pp = rotate_2D(x1p, y1p, theta)
+        x0pp += x0
+        y0pp += y0
+        x1pp += x0
+        y1pp += y0
+        out.append([[x0pp, y0pp], [x1pp, y1pp]])
+    return out
+
+def make_graph_edges(graph, pos):
+    """
+    Given a molecule graph generate starting and stop
+    points for the edges taking into account the bond
+    orders.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        Graph with bond oder attribute
+    pos: np.ndarray((2, len(graph)))
+        2D positions for the nodes
+    spacing:
+    sep: float
+
+    Returns
+    -------
+    list, list, dict
+        list of normal edges, aromatic edgs, and a dict of
+        all simple edges
+    """
```

Comment:
```suggestion
def make_graph_edges(graph, pos):
    """
    Given a molecule graph generate starting and stop
    points for the edges taking into account the bond
    orders.

    Parameters
    ----------
    graph: nx.Graph
        Graph with bond oder attribute
    pos: np.ndarray((2, len(graph)))
        2D positions for the nodes

    Returns
    -------
    list, list, dict
        list of normal edges, aromatic edgs, and a dict of
        all simple edges
    """
```
Consider giving this function a `**kwargs` to pass to `make_edge`